driver/src/contracts/stablex_contract.rs

```diff
@@ -0,0 +1,152 @@
+#[cfg(test)]
+extern crate mock_it;
+
+use dfusion_core::models::{AccountState, Order};
+
+// TODO - uncomment when removing "unimplemented!()"
+//use web3::contract::Options;
+//use web3::futures::Future;
+use web3::types::{H160, U128};
+
+use crate::error::DriverError;
+
+use super::base_contract::BaseContract;
+
+use std::env;
+use std::fs;
+
+type Result<T> = std::result::Result<T, DriverError>;
+
+#[allow(dead_code)]
+struct StableXContractImpl {
+    base: BaseContract
+}
+
+#[allow(dead_code)]
+impl StableXContractImpl {
+    pub fn new() -> Result<Self> {
+        let contract_json = fs::read_to_string("dex-contracts/build/contracts/StablecoinConverter.json").unwrap();
+        let address = env::var("STABLEX_CONTRACT_ADDRESS").unwrap();
+        Ok(
+            StableXContractImpl {
+                base: BaseContract::new(address, contract_json).unwrap()
+            }
+        )
+    }
+}
+
+pub trait StableXContract {
+    fn get_current_auction_index(&self) -> Result<u32>;
+    fn get_auction_data(&self, _index: u32) -> Result<(AccountState, Vec<Order>)>;
+
+    fn submit_solution(
+        &self,
+        _batch_index: u32,
+        _owners: Vec<H160>,
+        _order_ids: Vec<u16>,
+        _volumes: Vec<U128>,
+        _prices: Vec<U128>,
+        _token_ids_for_price: Vec<u16>,
+    ) -> Result<()>;
+}
+
+impl StableXContract for StableXContractImpl {
+    fn get_current_auction_index(&self) -> Result<u32> {
+        unimplemented!();
+//        self.base.contract
```

Comment:
Can we keep this commit clean and remove the code that will be added later?

Reply:
Yes sure,